src/OrchardCore/OrchardCore.DisplayManagement/Implementation/DefaultHtmlDisplay.cs

```diff
@@ -50,6 +50,16 @@ public async Task<IHtmlContent> ExecuteAsync(DisplayContext context)
             // Check if the shape is pre-rendered.
             if (shape is IHtmlContent htmlContent)
             {
+                if (htmlContent is PositionWrapper wrapper)
+                {
+                    while (wrapper.Value is PositionWrapper)
```

Comment:
Should this be an if statement or a while loop?

Also change it to 
(wrapper.Value is PositionWrapper positionWrapper)  

So you don't have to cast again

Reply:
no need to re define new variable `positionWrapper`, wrapper is already defined in if statement, i am reusing same var

Reply:
I see what you are suggesting! - updated